swadm/src/compliance.rs

```diff
@@ -0,0 +1,275 @@
+                    .await
+            }
+            LinkAction::Setup {
+                speed,
+                fec,
+                autoneg,
+                all,
+            } => compliance_links_setup(client, speed, fec, autoneg, all).await,
+        },
+    }
+}
+
+async fn compliance_links_list(
+    client: &Client,
+    pattern: Option<&str>,
+    all: bool,
+) -> anyhow::Result<()> {
+    let links = get_matching_links(client, pattern, all).await?;
+
+    let mut tw = TabWriter::new(stdout());
+    writeln!(
+        tw,
+        "{}\t{}\t{}",
+        "LINK".underline(),
+        "ENABLED?".underline(),
+        "STATE".underline()
+    )?;
+
+    for link in links {
+        writeln!(tw, "{}\t{}\t{}", link, link.enabled, link.link_state)?;
+    }
+
+    tw.flush().map_err(|e| e.into())
+}
+
+async fn compliance_links_enable(
+    client: &Client,
+    pattern: Option<&str>,
+    enabled: bool,
+    all: bool,
+) -> anyhow::Result<()> {
+    let links = get_matching_links(client, pattern, all).await?;
+
+    for link in links {
+        client
+            .link_enabled_set(&link.port_id, &link.link_id, enabled)
+            .await
+            .context(format!(
+                "failed to {} link {}",
+                if enabled { "enable" } else { "disable" },
+                link
+            ))?;
+        println!(
+            "{} link {}",
+            if enabled { "Enabled" } else { "Disabled" },
+            link
+        );
+    }
+
+    Ok(())
+}
+
+async fn get_matching_links(
+    client: &Client,
+    pattern: Option<&str>,
+    all: bool,
+) -> anyhow::Result<Vec<types::Link>> {
+    // Determine server-side filter
+    let server_filter = if all {
+        // If --all is specified, send user's pattern to server (or None for all links)
+        pattern
+    } else {
+        // If --all is not specified, get qsfp links from server
+        Some("qsfp")
+    };
+
+    let links = client
+        .link_list_all(server_filter)
+        .await
+        .context("failed to list links")?
+        .into_inner();
+
+    // Apply client-side filtering if needed
+    let filtered_links = if !all && pattern.is_some() {
+        // Filter the qsfp links by user's pattern
+        let user_pattern = pattern.unwrap();
+        links
+            .into_iter()
+            .filter(|link| link.to_string().contains(user_pattern))
+            .collect()
+    } else {
+        // Server filtering was sufficient
+        links
+    };
+
+    Ok(filtered_links)
+}
+
+async fn compliance_links_setup(
+    client: &Client,
+    speed: PortSpeed,
+    fec: PortFec,
+    autoneg: bool,
+    all: bool,
+) -> anyhow::Result<()> {
+    // Get all switch ports
+    let all_ports = client
+        .port_list()
+        .await
+        .context("failed to list switch ports")?
+        .into_inner();
+
+    // Filter to qsfp ports only unless --all is specified
+    let switch_ports: Vec<types::PortId> = if all {
```

Comment:
I believe dendrite automatically creates backplane links. I can see this being useful if we need to delete and then recreate backplane links for compliance, but I don't see the corresponding link deletion code?

Reply:
Dendrite does automatically create backplane links.  I didn't think we needed/wanted to be able to delete links, just disable them.  Happy to add a corresponding teardown command though with the same filtering semantics

Reply:
I'd like to keep the `--all` flag since I want rfk to be able to manage the backplane links just in case the need arises.  I don't anticipate he'll need to, but the tool should allow for it IMO